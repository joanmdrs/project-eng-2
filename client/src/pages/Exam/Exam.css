.container-exam {
    display: flex;
    background-color: #d0d6db;
    min-height: 100vh;
}

.box-exam {
    width: 100%;
}

.box-header-exam {
    padding: 10px;
    padding-bottom: 5px;
    background-color: #fff;
}

.box-icon-and-title {
    display: flex;
    align-items: baseline;
    gap: 10px;
}

.box-list-functions {
    color: #848484;
    font-size: 18px;
}

.box-icon-and-title .icon {
    font-size: 25px;
}

.box-exam .form-exam {
    padding: 30px;
    margin: 10px 15px 15px 15px;
    border: 1px solid #d0d6db;
    border-radius: 4px;
    background-color: #fff;
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
}



.input-textarea {
    height: 180px;
    
}



.col-btn-save-cancel {
    display: flex;
    gap: 10px;
    align-items: flex-end;
    justify-content: flex-end;
}

.btn-save, .btn-cancel {
    /* background-color: #fff ; */
}