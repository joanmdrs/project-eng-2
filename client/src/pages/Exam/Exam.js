import React, { useEffect, useState } from "react";
import Nav from "../../components/Nav/Nav";
import { Container as ContainerExam} from "../../components/Container/Container";
import { Box as BoxExam } from "../../components/Box/Box";
import { Header as HeaderExam} from "../../components/Header/Header"
import FormExam from "../../components/Forms/FormExam/FormExam";
import { addExam, deleteExam, getActionForm, getExams, getValuesInput, messageFailure, messagePrepareToUpdate, messageSucess, setFields, updateExam } from "../../services/ExamServices";
import ListExam from "../../components/ListExam/ListExam";
import Swal from 'sweetalert2';

export default function Exam(){

    const [listExams, setListExams] = useState([]);

    // POST OR UPDATE
    const handleSaveButton = () => {

        const data = getValuesInput();

        const action = getActionForm();

        if(action === "add"){
            try{
                addExam(data);
                messageSucess("New exam added.");
            }catch(error){
                messageFailure("Something went wrong.")
            }
        }else {
            try{
                updateExam(action, data);
                messageSucess("The informations about this exam were updated.")
            }catch(error){
                messageFailure("Something went wrong.")
            }  
        }
    }

    // LIST 
    // Após a renderização, retornar todos os exames 

    useEffect( () => {

        const fetchData = async () => {
            const data = await getExams();
            const exams = JSON.parse(data)
            setListExams(exams)
        }
    
        fetchData();
    }, []);
    
    // SETFIELDS  

    const handlePreparaToUpdate = (ExamID) => {

        let data = {}
        listExams.forEach(element => {
            element.id === ExamID ? data = element : data = data
        });

        setFields(data);
    
    }

    messagePrepareToUpdate(
        "Edit Exam",
        "Now you will edit this Exam's informations, be careful."
    );

    // DELETE 

    const handleDelete = (id) => {
        deleteExam(id);
    }


    return (
        <ContainerExam>
            <Nav />
            <BoxExam>
                <HeaderExam 
                    title="Exam" 
                    text="Exam registration: Include, Search, Change, Delete and List" 
                    
                ></HeaderExam>

                <FormExam handleSaveButton={handleSaveButton} />
                <ListExam 
                    exam={listExams} 
                    setFields={handlePreparaToUpdate} 
                    handleDelete={handleDelete} />
            </BoxExam>
        </ContainerExam>
    )
}