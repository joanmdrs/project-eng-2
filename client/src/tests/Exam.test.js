import renderer from 'react-test-renderer';
import Exam from "../pages/Exam/Exam";

it('testando a renderização a page Exam', () => {
  const component = renderer.create(Exam);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot(); 

  
})
