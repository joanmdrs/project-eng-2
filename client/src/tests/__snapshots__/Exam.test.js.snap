// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando a renderização a page Exam 1`] = `null`;
